### ISSUE TITLE
share:Data can be shared to un-followed devices.

### ISSUE THREAD

**Issue by gayatri678** [rbccps-iisc/ideam#176]

**Comment by arun-babu** [comment]
This is not a problem if both devices have the same owner.

Otherwise: We would definitely like to control how many shares can be made per minute or so.

**Comment by gayatri678** [comment]
The issue is, sharing and subscribing was happening successfully, in the absence of 'Follow' API. Our understanding was, 'share' api will work only after the 'follow' api is made.

**Comment by arun-babu** [comment]
Yes it is not always necessary to follow. Especially if both devices are owned by same user.

**Comment by dheepa68** [comment]
How does the user & device mapping happens? and is this check happening whenever a device under one user is trying to share with another device under the same user?

**Comment by arun-babu** [comment]
In ldap currently for each device there is a owner attribute.

If both devices have same owner then there is no need to follow. The owner can just call bind.
Else: Follow request must be sent. However, currently a owner1 can share without owner2 giving a follow request.

**Comment by dheepa68** [comment]
So, will a fix be provided for this current behavior?

**Comment by swaminathanvasanth** [comment]
Re-opening this as share is still permitted between two owners, without any follow request.
